ur_robot_driver/launch/ur_common.launch

```diff
@@ -19,13 +19,11 @@
   <arg name="tool_device_name" default="/tmp/ttyUR" doc="Local device name used for tool communication. Only used, when `use_tool_communication` is set to true."/>
   <arg name="tool_tcp_port" default="54321" doc="Port on which the robot controller publishes the tool comm interface. Only used, when `use_tool_communication` is set to true."/>
   <arg name="robot_description_file" doc="Robot description launch file."/>
-  <arg name="limited" default="false" doc="Use the description in limited mode (Every axis rotates from -PI to PI)"/>
   <arg name="headless_mode" default="false" doc="Automatically send URScript to robot to execute. On e-Series this does require the robot to be in 'remote-control' mode. With this, the URCap is not needed on the robot."/>
 
   <!-- robot model -->
   <include file="$(arg robot_description_file)">
-    <arg name="limited" value="$(arg limited)"/>
```

Comment:
would you also want to allow to pass `joint_limits_params` in order to be able to configure the former limited/unlimited variant?

Reply:
That doesn't exist any more. All joints get their full +-2pi range, except the elbow, as that is the only joint which can't actually use that range.

That ~solves~ works-around the issue sufficiently.

Reply:
I think, it's a valid question to raise what arguments should be forwarded / at what point users should write their own description launch files instead.

My initial thought:
 * **kinematics_params**: This is the most important one that will depend on the *actual robot *in use
* **joints_limits_params**: This might be changed depending on the use case (e.g. the robot being mounted on a table). I would naively assume that in this situation would provide their own xacro and thus their own launchfile. Therefor I wouldn't forward this argument.
* **physical_params**: This is only depending on the actual robot *model*, I don't see any necessity to forward this.
* **visual_params**: This is only depending on the actual robot *model*, I don't see any necessity to forward this.
* **transmission_hw_interface**: As far as I understand, this is used for gazebo, so I wouldn't forward it.
* **safety_limits**: Probably should be forwarded, see #198. Or is this something different?
* **safety_pos_margin**: see above
* **safety_k_position**: see above

Reply:
This seems to come down to how do we expect users to use the driver with custom setups (ie: just about any real use of the driver)?

Personally I would probably create a custom version of one of the `load_urNN.launch` files, override the `default`s for the relevant args and then make the bringup `.launch` file in `ur_robot_driver/launch` use *that* `.launch` file instead of the defaults.

That would only require the ability to pass a path to my custom `.launch` file to the bringup `.launch`.

It would not be very convenient to always have to provide *all* args to the bringup `.launch`.

> **visual_params**: This is only depending on the actual robot model, I don't see any necessity to forward this.

So I know of at least one use-case which would become possible by allowing access to this arg: ros-industrial/universal_robot#267, and this was actually partly what motivated me to suggest moving to a completely parameterised `_macro.xacro`: the ability to override specific aspects of the visual model of the robot.

Reply:
I think, I'll stick with the current implementation, as we actually want users to provide their own calibration file. If users want to modify other things themselves, they should bring along their own description launch file, as @gavanderhoorn suggested.

Documenting this still has to be done.

Reply:
Documentation added in f6a7328